### PR TITLE
Overwrite driver code to disable cqe_compression

### PR DESCRIPTION
**Pull request by dennisafa**

### CODE REVIEW

File: patches/disable_cqe_compression.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+@@ -690,12 +689,7 @@ mlx5_dev_spawn(struct rte_device *dpdk_dev,
+ 			mprq_caps.max_single_wqe_log_num_of_strides;
+ 	}
+ #endif
+-	if (RTE_CACHE_LINE_SIZE == 128 &&
+-	    !(dv_attr.flags & MLX5DV_CONTEXT_FLAGS_CQE_128B_COMP))
+-		cqe_comp = 0;
+-	else
+-		cqe_comp = 1;
+-	config->cqe_comp = cqe_comp;
```
Comment (taylor-stateless) [inline patches/disable_cqe_compression.patch:32]: I think you should have #ifdef DISABLE_CQE flags to toggle disabling it so we have a choice in the future wen compiling this things

**Review by taylor-stateless — CHANGES_REQUESTED** [review]
Add compilation flags to disable cqe.

**Review by OferMania — APPROVED** [review]

**Review by adickson66 — APPROVED** [review]

File: patches/disable_cqe_compression.patch [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+-	unsigned int cqe_comp;
+ 	unsigned int cqe_pad = 0;
+ 	unsigned int tunnel_en = 0;
+ 	unsigned int mpls_en = 0;
+@@ -690,12 +689,7 @@ mlx5_dev_spawn(struct rte_device *dpdk_dev,
+ 			mprq_caps.max_single_wqe_log_num_of_strides;
+ 	}
+ #endif
+-	if (RTE_CACHE_LINE_SIZE == 128 &&
+-	    !(dv_attr.flags & MLX5DV_CONTEXT_FLAGS_CQE_128B_COMP))
```
Comment (james-munson) [inline patches/disable_cqe_compression.patch:28]: Totally taking your word for the meaning and applicability of these other attributes.

**Review by james-munson — APPROVED** [review]

File: patches/disable_cqe_compression.patch (hunk 1 above)
Comment (dennisafa) [reply patches/disable_cqe_compression.patch:32]: Andy covered it. Closing this PR

**Pull request closed without merging**